Drop redundant miscfiles_read_generic_certs(nsswitch_domain)

21sw-clayton opened the pull request:
This is allowed elsewhere.

```
$ sesearch -A --source syslogd_t --target cert_t --perm read
allow nsswitch_domain cert_t:dir { getattr ioctl lock open read search };
allow nsswitch_domain cert_t:file { getattr ioctl lock map open read };
allow nsswitch_domain cert_t:lnk_file { getattr read };
allow syslogd_t non_security_file_type:dir { getattr ioctl lock open read search }; [ logging_syslogd_list_non_security_dirs ]:True

$ getsebool logging_syslogd_list_non_security_dirs
logging_syslogd_list_non_security_dirs --> off
```

zpytela commented:
nsswitch_domain is an attribute with 500+ domains assigned to it, so the removed line is far from being redundant

21sw-clayton commented:
Hi @zpytela

The `sesearch` result is not dependent on the value of `authlogin_nsswitch_use_ldap`, suggesting it is allowed elsewhere.

I had trouble specifically tracking that down, but I believe it is here: https://github.com/fedora-selinux/selinux-policy/blob/v42.19/policy/modules/system/authlogin.if#L451

zpytela commented:
No output line of the sesearch command is a subset of any other.

21sw-clayton commented:
Hi, @zpytela

I've found the source for `syslogd_t`: https://github.com/fedora-selinux/selinux-policy/commit/66742ce1af1d752fbefe8a512f16c9165634d07f

Here are more specific queries that I believe demonstrate redundancy:
```
$ sesearch -A -s nsswitch_domain -ds -t cert_t -dt --perm read
allow nsswitch_domain cert_t:dir { getattr ioctl lock open read search };
allow nsswitch_domain cert_t:file { getattr ioctl lock map open read };
allow nsswitch_domain cert_t:lnk_file { getattr read };

$ getsebool authlogin_nsswitch_use_ldap
authlogin_nsswitch_use_ldap --> off
```

21sw-clayton committed:
Drop redundant miscfiles_read_generic_certs(nsswitch_domain)

This is allowed elsewhere.

$ sesearch -A -s nsswitch_domain -ds -t cert_t -dt --perm read
allow nsswitch_domain cert_t:dir { getattr ioctl lock open read search };
allow nsswitch_domain cert_t:file { getattr ioctl lock map open read };
allow nsswitch_domain cert_t:lnk_file { getattr read };

$ getsebool authlogin_nsswitch_use_ldap
authlogin_nsswitch_use_ldap --> off

Signed-off-by: Clayton Casciato <ccasciato@21sw.us>

zpytela commented:
Can you be more specific which line is redundant?

Take into account also some other nsswitch_domain, e.g. snmpd_t, and other non_security_file_type, like cupsd_etc_t.
